internal/resolver/passthrough/passthrough.go

```diff
@@ -20,13 +20,20 @@
 // name without scheme back to gRPC as resolved address.
 package passthrough
 
-import "google.golang.org/grpc/resolver"
+import (
+	"errors"
+
+	"google.golang.org/grpc/resolver"
+)
 
 const scheme = "passthrough"
 
 type passthroughBuilder struct{}
 
 func (*passthroughBuilder) Build(target resolver.Target, cc resolver.ClientConn, opts resolver.BuildOptions) (resolver.Resolver, error) {
+	if target.Endpoint == "" && opts.Dialer == nil {
```

Comment:
I feel like we could do even stronger validation here (verify it could work when sent to `net.Dial`, e.g. split host/port & parse IP), but this is an improvement anyway.